### PR TITLE
Add firefox developer edition

### PR DESCRIPTION
**Pull request by danstewart**
#405

### CODE REVIEW

File: plugins/firefoxdev.plugin/install.sh [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+[Desktop Entry]
+Version=1.0
+Encoding=UTF-8
+Name=Firefox Dev
+Comment=Firefox Developer Edition
+Exec=/opt/firefox/firefox
+Icon=/opt/firefox/browser/icons/mozicon128.png
+Type=Application
+StartupNotify=true
+Keywords=firefox;dev;browser
```
Comment (bddenhartog) [inline plugins/firefoxdev.plugin/install.sh]: Change `dev` => `developer`

File: plugins/firefoxdev.plugin/metadata.json [hunk 2]
```diff
@@ -0,0 +1,18 @@
+{
+	"icon": "firefox",
```
Comment (bddenhartog) [inline plugins/firefoxdev.plugin/metadata.json]: Is the icon really just named `firefox`? How is the mainline edition differentiated here?

File: plugins/firefoxdev.plugin/uninstall.sh [hunk 3]
```diff
@@ -0,0 +1,5 @@
+#!/bin/bash
+
+rm -rf "/usr/bin/firefox-dev"
```
Comment (bddenhartog) [inline plugins/firefoxdev.plugin/uninstall.sh]: `-r` option is unnecessary here, this is a symlink (not a directory). Also be sure to change the path to `/usr/bin/firefox-developer` when the renaming is done elsewhere.

File: plugins/firefoxdev.plugin/uninstall.sh [hunk 4]
```diff
@@ -0,0 +1,5 @@
...
+rm -rf "/opt/firefox"
+rm -rf "/usr/local/share/applications/firefox-dev.desktop"
```
Comment (bddenhartog) [inline plugins/firefoxdev.plugin/uninstall.sh]: `-r` option is unecessary. Be sure to rename the file when it is renamed elsewhere.

**Comment by danstewart** [comment]
@bddenhartog Cheers - I've added the requested changes.

**Comment by bddenhartog** [comment]
> Ah wait, haven't checked the metadata icon - how do I find the correct icon name?
Tried firefox-developer, firefox-dev, & firefox-developer-edition but no luck.

The icon displayed in the GUI uses the `icon` attribute from `metadata.json` as the relative filename (without a prefix) for the image that's displayed for the plugin.

See [app.js:487-513](https://github.com/folkswithhats/fedy/blob/master/app.js#L487-L513) for more information about how this is implemented.

Basically, you'll want to an an icon named `firefox-developer{,.svg,.png}` to the plugin directory. Be sure to have git-lfs installed and initialized. The project is already configured to track `.png` files (see [`.gitattributes`](https://github.com/folkswithhats/fedy/blob/master/.gitattributes)), so you'll need to run `git lfs track '*.svg'` if you add the icon with a`.svg` extension.

File: plugins/firefoxdev.plugin/install.sh [hunk 5]
```diff
@@ -0,0 +1,26 @@
+#!/bin/bash
+
+CACHEDIR="/var/cache/fedy/firefoxdeveloper"
```
Comment (bddenhartog) [inline plugins/firefoxdev.plugin/install.sh]: Rename to `firefox-developer` to match the plugin name.

File: plugins/firefoxdev.plugin/install.sh [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+FILE="firefoxdeveloper.tar.bz2"
```
Comment (bddenhartog) [inline plugins/firefoxdev.plugin/install.sh]: Rename to `firefox-developer.tar.bz2`.

**Comment by danstewart** [comment]
@bddenhartog Should be all sorted now. Cheers.

**Pull request merged**